community-membership.md

```diff
@@ -263,7 +263,7 @@ before being able to contribute effectively.
 [Kubernetes GitHub Admin team]: /github-management/README.md#github-administration-team
 [Kubernetes GitHub organizations]: /github-management#actively-used-github-organizations
 [Kubernetes org]: https://github.com/kubernetes
-[kubernetes-dev@googlegroups.com]: https://groups.google.com/forum/#!forum/kubernetes-dev
+[dev@kubernetes.io]: https://groups.google.com/forum/#!forum/kubernetes-dev
```

Comment:
This link needs to be changed to the new destination

Reply:
Hey @cblecker it would really helpful if you can please provide link for new forum destination as I am not able to find it for dev@kubernetes.io